bokeh/models/legends.py

```diff
@@ -0,0 +1,208 @@
+""" Renderers for various kinds of legends that can be added to
+Bokeh plots
+
+"""
+from __future__ import absolute_import
+
+from ..core.enums import Orientation, LegendLocation
+from ..core.property_mixins import LineProps, FillProps, TextProps
+from ..core.properties import (
+    Int, String, Enum, Instance, List, Dict, Tuple, Include, Either,
+    Float, Override,
+)
+
+from .formatters import TickFormatter, BasicTickFormatter
+from .mappers import ColorMapper
+from .renderers import GuideRenderer, GlyphRenderer
+from .tickers import Ticker, BasicTicker
+
+class Legend(GuideRenderer):
```

Comment:
The refactoring of `Legend` into a subclass of `GuideRenderer` is isolated in PR #4851 and this branch is merged off of the #4851 parent. I should have mentioned that in the initial comment (I'll add it now).